osu.Game/Rulesets/UI/Scrolling/ScrollingHitObjectContainer.cs

```diff
@@ -150,81 +147,41 @@ private void flipPositionIfRequired(ref float position)
             }
         }
 
-        protected override void OnAdd(DrawableHitObject drawableHitObject) => onAddRecursive(drawableHitObject);
-
-        protected override void OnRemove(DrawableHitObject drawableHitObject) => onRemoveRecursive(drawableHitObject);
-
-        private void onAddRecursive(DrawableHitObject hitObject)
+        protected override void OnAdd(DrawableHitObject drawableHitObject)
         {
-            invalidateHitObject(hitObject);
-
-            hitObject.DefaultsApplied += invalidateHitObject;
-
-            foreach (var nested in hitObject.NestedHitObjects)
-                onAddRecursive(nested);
```

Comment:
Why was the recursion wrt nested objects removed here? I'm not seeing it explained anywhere, and it seems like a pretty big thing to remove.

Reply:
This change is not part of this PR. It is included in #13257.